tests/backup/backup_restore_basic_test.go

```diff
@@ -2763,3 +2764,446 @@ var _ = Describe("{BackupRestoreOnDifferentK8sVersions}", func() {
 		CleanupCloudSettingsAndClusters(backupLocationMap, cloudCredName, cloudCredUID, ctx)
 	})
 })
+
+// BackupCRsThenMultipleRestoresOnHigherK8sVersion deploys CRs (CRD + webhook) -> backups them up -> creates two simulatanous restores on a cluster with higher K8s version :: one restore is Success and other PartialSuccess
+var _ = Describe("{BackupCRsThenMultipleRestoresOnHigherK8sVersion}", func() {
+
+	var (
+		backupNames          []string             // backups in px-backup
+		restoreNames         []string             // restores in px-backup
+		restoreLaterNames    []string             // restore-laters in px-backup
+		scheduledAppContexts []*scheduler.Context // Each Context is for one Namespace which corresponds to one App
+		preRuleNameList      []string
+		postRuleNameList     []string
+		scheduledClusterUid  string
+		cloudCredName        string
+		cloudCredUID         string
+		backupLocationUID    string
+		backupLocationName   string
+	)
+
+	var (
+		appList           = Inst().AppList
+		namespaceMapping  = make(map[string]string)
+		backupLocationMap = make(map[string]string)
+		labelSelectors    = make(map[string]string)
+		providers         = getProviders()
+	)
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("BackupCRsThenMultipleRestoresOnHigherK8sVersion", "deploy CRs (CRD + webhook); then backup; create two simulatanous restores on cluster with higher K8s version; one restore is Success and other PartialSuccess", nil, 83716)
+	})
+
+	It("Deploy CRs (CRD + webhook); Backup; two simulatanous Restores with one Success and other PartialSuccess. (Backup and Restore on different K8s version)", func() {
+
+		defer func() {
+			log.InfoD("switching to default context")
+			err := SetClusterContext("")
+			log.FailOnError(err, "failed to SetClusterContext to default cluster")
+		}()
+
+		Step("Verify if app used to execute test is a valid/allowed spec (apps) for *this* test", func() {
+			log.InfoD("specs (apps) allowed in execution of test: %v", appsWithCRDsAndWebhooks)
+			for i := 0; i < len(appList); i++ {
+				contains := Contains(appsWithCRDsAndWebhooks, appList[i])
+				dash.VerifyFatal(contains, true, fmt.Sprintf("app [%s] allowed in execution of this test", appList[i]))
+			}
+		})
```

Comment:
Firstly, this logic will fail if multiple apps are passed and the `appList[0]` odes not contain an app with CRDs.

Secondly, for this test case you need make sure you manipulate the `Inst().AppList` by mentioning

```
Inst().AppList = []string{"elasticsearch-crd-webhook"}
```

If you do this, you don't have to bother about the AppList passed.

Reply:
Rather than manipulating the app list, wouldn't it be better to filter the list of app input by the user, such that we only use apps which are permitted. This way:
1. any future tests can use that special list to filter stuff out
2. this test can also be run with other tests with similar requirements, without this test manipulating the applist for every other test.
I made a change, you can check it out, when it's commited